hlsdl.go

```diff
@@ -72,96 +76,63 @@ func wait(wg *sync.WaitGroup) chan bool {
 }
 
 func (hlsDl *HlsDl) downloadSegment(segment *Segment) error {
-	req, err := newRequest(segment.URI, hlsDl.headers)
-	if err != nil {
-		return err
-	}
-
-	res, err := hlsDl.client.Do(req)
-	if err != nil {
-		return err
-	}
-	defer res.Body.Close()
-
-	if res.StatusCode != 200 {
-		return errors.New(res.Status)
-	}
-
-	file, err := os.Create(segment.Path)
-	if err != nil {
-		return err
-	}
-	defer file.Close()
-
-	if _, err := io.Copy(file, res.Body); err != nil {
-		return err
-	}
-
-	return nil
+	hlsDl.client.SetRetryCount(5).SetRetryWaitTime(time.Second)
+	_, err := hlsDl.client.R().SetHeaders(hlsDl.headers).SetOutput(segment.Path).Get(segment.URI)
```

Comment:
@yukunwu Does Resty check the httpCode here?

Reply:
No,I will fix it.